docs/development-crds.md

```diff
@@ -0,0 +1,166 @@
+
+```bash
+$ kubectl describe crd tests.tests.testkube.io
+Name:         tests.tests.testkube.io
+Namespace:    
+Labels:       app.kubernetes.io/managed-by=Helm
+Annotations:  controller-gen.kubebuilder.io/version: v0.4.1
+              meta.helm.sh/release-name: testkube
+              meta.helm.sh/release-namespace: testkube
+API Version:  apiextensions.k8s.io/v1
+Kind:         CustomResourceDefinition
+...
+```
+
+Below, you will find short descriptions and example declarations of the custom resources defined by Testkube.
+
+## Tests
+
+Testkube Tests can be defined as a single executable unit of tests. Depending on the test type, this can mean one or multiple test files.
+
+To get all the test types available in your cluster, check the executors:
+
+```bash
+$ kubectl testkube get executors -o yaml | grep -A1 types
+    types:
+    - postman/collection
+--
+    types:
+    - curl/test
+--
+    types:
+    - cypress/project
+--
+    types:
+    - k6/script
+--
+    types:
+    - postman/collection
+--
+    types:
+    - soapui/xml
+```
+
+When creating a Testkube Test, there are multiple supported input types:
+
+* string
+* Git directory
+* Git file
+* file URI
+
+Variables can be configured using the `variables` field as shown below.
+
+```yml
+apiVersion: tests.testkube.io/v2
+kind: Test
+metadata:
+  name: example-test
+  namespace: testkube
+spec:
+  content:
+    data: "{...}"
+    type: string
+  type: postman/collection
+  variables:
+    var1:
+      name: var1
+      type: basic
+      value: val1
+    sec1:
+      name: sec1
+      type: secret
+      valueFrom:
+        secretKeyRef:
+          key: sec1
+          name: vartest4-testvars
+```
+
+## Test Suites
+
+Testkube Test Suites are collections of Testkube Tests of the same or different types.
+
+```yml
+apiVersion: tests.testkube.io/v1
+kind: TestSuite
+metadata:
+  name: example-testsuite
+  namespace: testkube
+spec:
+  description: Example Test Suite
+  steps:
+    - execute:
+        name: example-test1
+        namespace: testkube
+    - delay:
+        duration: 1000
+    - execute:
+        name: example-test2
+        namespace: testkube
+```
+
+## Executors
+
+Executors are Testkube-specific test runners. There are a list of predefined Executors coming with Testkube. You can also write your own custom Testkube Executor using [this guide](https://kubeshop.github.io/testkube/executor-custom/).
+
+Example:
+
+```yml
+apiVersion: executor.testkube.io/v1
+kind: Executor
+metadata:
+  name: example-executor
+  namespace: testkube
+spec:
+  executor_type: job  
+  image: YOUR_USER/testkube-executor-example:1.0.0 
+  types:
+  - example/test      
+  content_types:
+  - string
+  - file-uri
+  - git-file
+  - git-dir
+  features: 
+  - artifacts
+  - junit-report
+```
+
+## Webhooks
+
+Testkube Webhooks are event-based configurable scripts. They are executed when a test is either started or finished. This can be defined under `events`.
```

Comment:
I would not describe them as scripts as it could suggest that user can do anything with them 

Currently it's only standard web hook so only http post call with execution object as payload with current state of execution

Reply:
Updated this as well